### ISSUE TITLE
Retain old direct hw access callback based audio for SDL3 as an option

### ISSUE THREAD

**Issue by ell1e** [libsdl-org/SDL#8074]
I wanted to heavily suggest you retain old direct hw access callback based audio for SDL3 as an option after seeing Ryan's video on the new audio system WITHOUT a mixer, buffer chain, "virtual" device forced in between. My apologies in advance since I might have misunderstood how it works. But it seems to me from that video that SDL3 forces a virtual device with a virtual mixer in between with another buffer chain, even when using the "traditional" callback API. If true, for properly written high performance audio apps geared for low latency audio this seems like a universal downgrade and very undesirable. I also find it a bit surprising to have this in the core lib at all, these mixers, what happened to having them in useful separate libs like SDL_mixer only for those users who actually want to stack a software mixer into their SDL setup? It seems potentially a worrying approach to me to force all this additional complexity into the core when many won't use it and still need everyone to ship it. Sorry again if I misunderstood some of the new systems from the info I saw.

**Comment by ell1e** [comment]
Thanks so much for the quick response! For what it's worth when you say "almost equivalent", I'm almost purely just concerned with additional buffer copies. A stream object with mixer support sounds like it would use its own buffer to me before going to the OS API buffer, which to me sounds rather bad. (Whether DirectSound then adds another copy on top isn't of my concern since that seems unavoidable, but copying added by SDL3 is not.) I didn't quite gather from your response if SDL3 adds more copying, it just seemed to me like it naturally might need to for a mixer shoved in.

**Comment by icculus** [comment]
It might add an extra copy if necessary, but it doesn't add _latency_ as we feed the audio device the same way internally that we did in SDL2.

And if there's a single device and a single stream, which would be normal in a game that provides its own mixing, SDL skips the mixing step and just provides the single source of data directly to the backend.

tl;dr: I think we're probably okay here.

**Comment by ell1e** [comment]
An extra copy always adds latency which is why I brought it up, so I think that is undesirable.
